doc/user/writing_spec.md

```diff
@@ -0,0 +1,45 @@
+# Writing Cluster Spec
+ 
+Users tell operator the desired state of an etcd cluster via spec in `Cluster` third party resource (TPR).
+See [example](../../example/example-etcd-cluster.yaml).
+
+Users might wonder what to specify in the spec.
```

Comment:
I think we should remove line6 to 13.

replace it with a markdown table similar to https://github.com/coreos/prometheus-operator/blob/master/Documentation/prometheus.md#specification